packages/shared/src/baduk/board/BadukBoardAbstract.ts

```diff
@@ -0,0 +1,55 @@
+import type { IBadukBoard } from "../../Interfaces/IBadukBoard";
+import type { BadukWithAbstractBoardConfig } from "../badukWithAbstractBoard";
+import { BoardPattern } from "../badukWithAbstractBoard";
+import { Intersection } from "./intersection";
+import { PolygonalBoardHelperFunctions } from "./PolygonalBoardHelper";
+import { Vector2D } from "./Vector2D";
+
+export class BadukBoardAbstract implements IBadukBoard 
+{
+	Intersections: Intersection[];
+	Array2D: Intersection[][] | null;
+
+	constructor(conf: BadukWithAbstractBoardConfig)
+	{
+		switch (conf.pattern)
+		{
+			case BoardPattern.Rectangular:
```

Comment:
Have you tested this path? I get an error when passing in BoardPattern.Rectangular:

```
➤ YN0000:     TypeError: Cannot read property 'push' of undefined
➤ YN0000: 
➤ YN0000:       25 |
➤ YN0000:       26 |                                            this.Intersections.push(intersection);
➤ YN0000:     > 27 |                                            this.Array2D[i].push(intersection);
➤ YN0000:          |                                                            ^
➤ YN0000:       28 |
➤ YN0000:       29 |                                            if (i > 0) {
➤ YN0000:       30 |                                                    intersection.ConnectTo(this.Array2D[i - 1][j], true);
➤ YN0000: 
➤ YN0000:       at new BadukBoardAbstract (src/baduk/board/BadukBoardAbstract.ts:27:23)
➤ YN0000:       at new BadukWithAbstractBoard (src/baduk/badukWithAbstractBoard.ts:42:22)
```

Here is the test code I ran to get this error:

```
test("Rectangular", () => {
  const game = new BadukWithAbstractBoard({
    width: 19,
    height: 19,
    komi: 5.5,
    pattern: BoardPattern.Rectangular,
  });
  expect(game.exportState).toBe("ignore this for now");
});
```

Reply:
No I did not test this, sorry. I'll debug it when I have some time.

I might have added a lot of unnecessary code, where I wasn't sure myself where its going.

Reply:
No worries, just wanted to check to see if I was using the wrong config.  Rectangular support isn't a huge priority bc `baduk` already does that (although it is cool the graph can represent arbitrary configurations!)